core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -382,27 +378,43 @@ private int assignNewColumnId() {
   }
 
   private TableMetadata applyChangesToMapping(TableMetadata metadata) {
```

Comment:
This method has a very specific name as it was previously only handling updates to the name mapping. I think we should rename it into something more generic now. What about just `applyChanges`?